pkg/gather/clusterconfig/clusterconfig_gatherer.go

```diff
@@ -0,0 +1,94 @@
+	"image_pruners":                     common.ImportantFunc((*Gatherer).GatherClusterImagePruner),
+	"feature_gates":                     common.ImportantFunc((*Gatherer).GatherClusterFeatureGates),
+	"oauths":                            common.ImportantFunc((*Gatherer).GatherClusterOAuth),
+	"ingress":                           common.ImportantFunc((*Gatherer).GatherClusterIngress),
+	"proxies":                           common.ImportantFunc((*Gatherer).GatherClusterProxy),
+	"certificate_signing_requests":      common.ImportantFunc((*Gatherer).GatherCertificateSigningRequests),
+	"crds":                              common.ImportantFunc((*Gatherer).GatherCRD),
+	"host_subnets":                      common.ImportantFunc((*Gatherer).GatherHostSubnet),
+	"machine_sets":                      common.ImportantFunc((*Gatherer).GatherMachineSet),
+	"install_plans":                     common.ImportantFunc((*Gatherer).GatherInstallPlans),
+	"service_accounts":                  common.ImportantFunc((*Gatherer).GatherServiceAccounts),
+	"machine_config_pools":              common.ImportantFunc((*Gatherer).GatherMachineConfigPool),
+	"container_runtime_configs":         common.ImportantFunc((*Gatherer).GatherContainerRuntimeConfig),
+	"netnamespaces":                     common.ImportantFunc((*Gatherer).GatherNetNamespace),
+	"openshift_apiserver_operator_logs": common.FailableFunc((*Gatherer).GatherOpenShiftAPIServerOperatorLogs),
+	"openshift_sdn_logs":                common.FailableFunc((*Gatherer).GatherOpenshiftSDNLogs),
+	"openshift_sdn_controller_logs":     common.FailableFunc((*Gatherer).GatherOpenshiftSDNControllerLogs),
+	"openshift_authentication_logs":     common.FailableFunc((*Gatherer).GatherOpenshiftAuthenticationLogs),
+	"sap_config":                        common.FailableFunc((*Gatherer).GatherSAPConfig),
+	"sap_license_management_logs":       common.FailableFunc((*Gatherer).GatherSAPVsystemIptablesLogs),
+	"sap_pods":                          common.FailableFunc((*Gatherer).GatherSAPPods),
+	"sap_datahubs":                      common.FailableFunc((*Gatherer).GatherSAPDatahubs),
+	"olm_operators":                     common.FailableFunc((*Gatherer).GatherOLMOperators),
+	"pod_network_connectivity_checks":   common.FailableFunc((*Gatherer).GatherPNCC),
+}
+
+func New(
+	gatherKubeConfig, gatherProtoKubeConfig, metricsGatherKubeConfig *rest.Config,
+	anonymizer *anonymization.Anonymizer,
+) *Gatherer {
+	return &Gatherer{
+		gatherKubeConfig:        gatherKubeConfig,
+		gatherProtoKubeConfig:   gatherProtoKubeConfig,
+		metricsGatherKubeConfig: metricsGatherKubeConfig,
+		anonymizer:              anonymizer,
+	}
+}
+
+func (g *Gatherer) GetName() string {
+	return "clusterconfig"
+}
+
+func (g *Gatherer) GetGatheringFunctions() map[string]common.GatheringClosure {
+	result := make(map[string]common.GatheringClosure)
+
+	for funcName, function := range gatheringFunctions {
+		function := function
```

Comment:
It looks like we don't need this line :)

Reply:
We actually do https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables otherwise all the closures will reference one function.